### PR TITLE
ADR-45 - Testing flow for long-running server tests

### PR DESCRIPTION
**Pull request by mprimi**

### CODE REVIEW

File: adr/ADR-45.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+However, addressing these would have the additional benefit of creating a new path for more long-running tests to be created in the future.
+
+Other tests that are currently slow but enabled could also be relocated, (e.g., `TestJetStreamClusterStreamOrphanMsgsAndReplicasDrifting` average 6+ minutes runtime).
+
+## Design
+
+Proposed change summary:
+
+ - Create a new test for long files in the server package, e.g.: `server/long_running_tests.go`
```
Comment (wallyqs) [inline adr/ADR-45.md:37]:
```suggestion
 - Create a new test for long files in the server package, e.g.: `server/long_running_test.go`
```

File: adr/ADR-45.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+|----------|------------|---------|----------------|
+| 1        | 2022-09-16 | @mprimi | Initial design |
+
+## Context and Problem Statement
+
+The server codebase contains dormant tests currently skipped because they take too long to execute during PR validation.
+
+## [Context | References | Prior Work]
+
+The list of currently skipped test is relatively short:
```
Comment (ReubenMathew) [inline adr/ADR-45.md:20]:
```suggestion
The list of currently skipped tests is relatively short:
```

**Review by ReubenMathew — CHANGES_REQUESTED** [review]

**Comment by ripienaar** [comment]
I think it's a useful approach and worth doing but not sure it merits an ADR - we tend to use these for big design things not for decisions

**Comment by scottf** [comment]
I'd like to see a test design doc instead of references to code, and I think that does belong here. I might remove the server category to keep it distinct as testing.

**Comment by mprimi** [comment]
> not sure it merits an ADR

Fair enough. I won't try to merge this PR. 
I wanted to collect some feedback/opinions, then I'll close it.

**Comment by mprimi** [comment]
GH Pipeline: https://github.com/nats-io/nats-server/actions/workflows/long-tests.yaml
BK Pipeline: https://buildkite.com/synadia/nats-server-long-running-tests
BK Analytics: https://buildkite.com/organizations/synadia/analytics/suites/nats-server-long-running-tests
PR: https://github.com/nats-io/nats-server/pull/5920

**Pull request closed without merging**